### ISSUE TITLE
Port FxCop rule CA1307: SpecifyStringComparison

### ISSUE THREAD

**Issue by lgolding** [dotnet/roslyn-analyzers#415]
**Title:** Specify StringComparison

**Description:**

A string comparison operation uses a method overload that does not set a StringComparison parameter.

**Dependency:** None

**Notes:**

**Comment by srivatsn** [comment]
@qinxgit plans to implement this.

**Comment by srivatsn** [comment]
Removing community tag since @qinxgit is not going to be able to get to this.

**Comment by genlu** [comment]
I have an prototype for this rule here:
https://github.com/genlu/roslyn-analyzers/commit/6382e0cdb2c9dad086eddb596037fceac6e5be0c

Anyone who picks up this work, please feel free to use that code :)

**Comment by basoundr** [comment]
#915 implements this rule

**Comment by mavasani** [comment]
Don't we need fixer for this rule?

**Comment by basoundr** [comment]
Yes, we do and thats why I have added the Needs-Fixer label

**Comment by mavasani** [comment]
We should keep the issue open right?

**Comment by basoundr** [comment]
Yes, we do. My mistake, I issued 'Comment & Close' instead of just 'Comment'

**Comment by kmytien** [comment]
Hi, I'm working with the code analyzers at the moment and noticed 1307 doesn't have a code fix implemented yet. Will there be one soon?
